docs/recitations/reci3-deployment.md

```diff
@@ -0,0 +1,180 @@
+
+# Recitation 3: Deployment
+
+  
+
+## Overview
+
+Throughout this recitation, students will:
+
+  
+
+* Understand the fundamental steps involved in deployment
+
+* Learn about CI/CD and the basics of how to set it up in a Github repo
+
+* Get hands on experience deploying with multiple cloud providers - Render and Google Cloud Platform.
+
+* Set up and get familiarized with the _Extreme Startup_ web app for use in Tuesday's (9/19) lecture
+
+  
+
+## Definitions
+
+  
+
+Before we begin, we have to understand our goals with CI/CD. It is a software development approach that utilizes frequent, automated testing and deployment to ensure that changes to the codebase can be rapidly and safely incorporated into the production environment.
+
+  
+
+**Continuous Integration (CI)** refers to the practice of automatically building and testing the codebase whenever new changes are introduced. This involves integrating code changes from multiple developers into a shared repository and running automated tests to ensure that the changes do not break the existing functionality.
+
+  
+
+**Continuous Deployment/Delivery (CD)** refers to the practice of automatically deploying code changes to the production environment after they have been tested and approved. Continuous Deployment involves automatically deploying changes to the production environment as soon as they pass automated tests, while Continuous Delivery involves deploying changes to a staging environment for further testing and approval before being deployed to production.
+
+  
+
+Together, CI/CD enables software teams to rapidly and safely develop, test, and deploy changes to the codebase, improving the speed and quality of software delivery while reducing the risk of errors and downtime.
+
+  
+
+There's quite a lot to learn about CI/CD. If you're interested in learning more, [click here!](https://about.gitlab.com/topics/ci-cd/)
+
+  
+
+## Task 0: Setting up your repository
+
+  
+
+For this recitation, we'll be working with a basic web app built on Next.js that responds to HTTP requests. Your job on Tuesday during the _Extreme Startup_ game will be to parse these requests and figure out what the correct response is. Today we'll focus on setting up the app so that everything is in place for the game.
+
+  
+
+The steps to do this are as follows:
+
+!!! Warning
+	Do not use codespaces for this task, and the rest of the recitation. Everything must be done locally.
+
+1. Fork this repo: [https://github.com/CMU-313/basic-web-app](https://github.com/CMU-313/basic-web-app).
+
+2. Follow the instructions in the readme to run and test the development server locally.
+  
```

Comment:
Ask them to go to https://localhost:3000 try out putting two different queries in the box:
"Who was Shakespeare?" and "What is your Andrew ID"?

Also ask them to run https://localhost:3000/api?q=hello and https://localhost:3000/api?q=shakespeare to see the API endpoint.